Source/FileManager/NamingTemplate/PropertyTagClass[TClass].cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace FileManager.NamingTemplate;
+
+public delegate string PropertyFormatter<T>(ITemplateTag templateTag, T value, string formatString);
+
+public class PropertyTagClass<TClass> : TagClass
+{
+	public PropertyTagClass(bool caseSensative = true) : base(typeof(TClass), caseSensative) { }
+
+	/// <summary>
+	/// Register a nullable value type property.
+	/// </summary>
+	/// <typeparam name="U">Type of the property from <see cref="TClass"/></typeparam>
+	/// <param name="propertyGetter">A Func to get the property value from <see cref="TClass"/></param>
+	/// <param name="formatter">Optional formatting function that accepts the <typeparamref name="U"/> property and a formatting string and returnes the value formatted to string</param>
+	public void RegisterProperty<U>(ITemplateTag templateTag, Func<TClass, U?> propertyGetter, PropertyFormatter<U> formatter = null)
+		where U : struct
+		=> RegisterPropertyInternal(templateTag, propertyGetter, formatter);
+
+	/// <summary>
+	/// Register a non-nullable value type property
+	/// </summary>
+	/// <typeparam name="U">Type of the property from <see cref="TClass"/></typeparam>
+	/// <param name="propertyGetter">A Func to get the property value from <see cref="TClass"/></param>
+	/// <param name="formatter">Optional formatting function that accepts the <typeparamref name="U"/> property and a formatting string and returnes the value formatted to string</param>
+	public void RegisterProperty<U>(ITemplateTag templateTag, Func<TClass, U> propertyGetter, PropertyFormatter<U> formatter = null)
+		where U : struct
+		=> RegisterPropertyInternal(templateTag, propertyGetter, formatter);
+
+	/// <summary>
+	/// Register a string type property.
+	/// </summary>
+	/// <param name="propertyGetter">A Func to get the string property from <see cref="TClass"/></param>
+	/// <param name="formatter">Optional formatting function that accepts the string property and a formatting string and returnes the value formatted to string</param>
+	public void RegisterProperty(ITemplateTag templateTag, Func<TClass, string> propertyGetter, PropertyFormatter<string> formatter = null)
+		=> RegisterPropertyInternal(templateTag, propertyGetter, formatter);
+
+	private void RegisterPropertyInternal(ITemplateTag templateTag, Delegate propertyGetter, Delegate formatter)
```

Comment:
Good call. The other name rubbed me the wrong way but I didn't want to nit pick.